public/app/core/components/NavBar/NavBarItem.test.tsx

```diff
@@ -1,55 +1,144 @@
 import React from 'react';
 import { render, screen } from '@testing-library/react';
-import userEvent from '@testing-library/user-event';
 import { BrowserRouter } from 'react-router-dom';
-import NavBarItem from './NavBarItem';
+import NavBarItem, { Props } from './NavBarItem';
+import userEvent from '@testing-library/user-event';
+
+const onClickMock = jest.fn();
+const defaults: Props = {
+  children: undefined,
+  link: {
+    text: 'Parent Node',
+    onClick: onClickMock,
+    children: [
+      { text: 'Child Node 1', onClick: onClickMock, children: [] },
+      { text: 'Child Node 2', onClick: onClickMock, children: [] },
+    ],
+  },
+};
+
+function getTestContext(overrides: Partial<Props> = {}) {
+  jest.clearAllMocks();
+  const props = { ...defaults, ...overrides };
+
+  const { rerender } = render(
+    <BrowserRouter>
+      <NavBarItem {...props}>{props.children}</NavBarItem>
+    </BrowserRouter>
+  );
+
+  return { rerender };
+}
 
 describe('NavBarItem', () => {
-  it('renders the children', () => {
-    const mockLabel = 'Hello';
-    render(
-      <BrowserRouter>
-        <NavBarItem label={mockLabel}>
-          <div data-testid="mockChild" />
-        </NavBarItem>
-      </BrowserRouter>
-    );
-
-    const child = screen.getByTestId('mockChild');
-    expect(child).toBeInTheDocument();
-  });
+  describe('when url property is not set', () => {
+    it('then it renders the menu trigger as a button', () => {
+      getTestContext();
+
+      expect(screen.getAllByRole('button')).toHaveLength(1);
+    });
+
+    describe('and clicking on the menu trigger button', () => {
+      it('then the onClick handler should be called', () => {
+        getTestContext();
+
+        userEvent.click(screen.getByRole('button'));
+
+        expect(onClickMock).toHaveBeenCalledTimes(1);
+      });
+    });
+
+    describe('and hovering over the menu trigger button', () => {
+      it('then the menu items should be visible', () => {
+        getTestContext();
+
+        userEvent.hover(screen.getByRole('button'));
+
+        expect(screen.getByText('Parent Node')).toBeInTheDocument();
```

Comment:
```suggestion
        expect(screen.getByRole('menuitem', { name: 'Parent Node' })).toBeInTheDocument();
```
I think we can test this and other menu items with more accessibility in mind, as they seem to have `menuitem` role.